core/utils/poly-look/src/visualisations/charts/bubble-charts/bubbleCluster/bubbleCluster.js

```diff
@@ -185,24 +309,28 @@ export class BubbleCluster extends Chart {
     const packLayout = this._pack();
     const root = packLayout(hierarchicalData);
     this._setUpFilters();
-    const nodes = this.chart
-      .selectAll("g")
-      .data(root.descendants().filter(({ parent }) => !!parent));
+    const nodes = this.chart.selectAll(".bubble-group").data(
+      root.descendants().filter(({ parent }) => !!parent),
+      (d) => d.data.category
+    );
     nodes.exit().remove();
-    this._updateBubbles(nodes);
     const newBubbleGroups = this._addNewBubbleGroups(nodes);
     this._addBubbles(newBubbleGroups);
+    this._updateBubbles(nodes);
 
     if (this._text) {
       this._addTextToBubbleGroup(newBubbleGroups);
       this._updateBubbleTexts(nodes);
 
-      // Bit of a hack to ensure that the texts of bubbles with children are
-      // rendered on top of those children. As a consequence of this, bubbles
-      // that have both a solid color fill and children will hide their
-      // children. If we were to render all texts on top of all bubbles, we
-      // could avoid this problem.
       newBubbleGroups.filter((d) => d.children).raise();
     }
+
+    if (this._label) {
+      let labelGroup = nodes.select(".label-group");
+      const empty = labelGroup.empty();
```

Comment:
If this is used only below, consider eliminating it. At any rate, it does not seem to be affected by the next line, so consider flipping these two lines.

Reply:
Yeah that's a logical bug :D Was from on old try

Reply:
Ah no, that's actually necessary as we need to see to which selection we append the labels.

Reply:
So `this._addLabel` is going to actually affect labelGroup, since both are acting on the DOM, right?